include/mxnet/lib_api.h

```diff
@@ -203,6 +214,16 @@ enum MXDType {
   kUNSET = 100,
 };
 
+/*!
+ * \brief Context info passing from MXNet OpContext
+ * dev_type is string repr of supported context, currently only "cpu" and "gpu"
+ * dev_id is the device index where the tensor locates
+ */
+typedef struct {
+  std::string dev_type;
```

Comment:
I worry about the ABI compatibility of std::string, since MXNet and custom lib may be built in different version of compilers or different compilers, e.g. gcc5 and gcc 9, gcc and clang. The implementation of std::string may be different. A solution is to use a C string char*.

Reply:
no, this string only serves as an easy way for custom library to check the device type, the actual context names passed from custom library to MXNet are still in const char *. Line 971 in lib_api.h is what we passed through the boundary

Reply:
I see. Thank you!

Reply:
I will appreciate if you can make another review!